File: multi-lang-api-demo-main/python/person_api_service/app/repository_interface.py
from abc import ABC, abstractmethod
from typing import Optional

from app.models import PersonCreate, PersonUpdate


class PersonRepositoryInterface(ABC):
    @abstractmethod
    def create_person(self, person: PersonCreate):
        pass

    @abstractmethod
    def get_person(self, person_id: int):
        pass

    @abstractmethod
    def update_person(self, person_id: int, person: PersonUpdate):
        pass

    @abstractmethod
    def delete_person(self, person_id: int) -> bool:
        pass

    @abstractmethod
    def search_by_name(self, first_name: Optional[str], last_name: Optional[str]):
        pass

    @abstractmethod
    def list_by_city_state(self, city: str, state: str):
        pass

File: multi-lang-api-demo-main/go/person_api_service/repository/repository_interface.go
package repository

import "github.com/example/person_api_service/models"

// PersonRepositoryInterface defines the operations for person data storage.
type PersonRepositoryInterface interface {
	CreatePerson(person models.PersonCreate) (*models.Person, error)
	GetPerson(id string) (*models.Person, error)
	UpdatePerson(id string, personData models.PersonUpdate) (*models.Person, error)
	DeletePerson(id string) (bool, error)
	SearchByName(firstName, lastName string) ([]models.Person, error)
	ListByCityState(city, state string) ([]models.Person, error)
}
